Web/src/games/BalloonShake.tsx

```diff
@@ -3,67 +3,93 @@ import {
   fromEvent, Observable, EMPTY,
 } from 'rxjs';
 import {
-  filter, map, throttleTime, pluck,
+  filter, map, pluck, debounceTime, startWith,
 } from 'rxjs/operators';
 import useCounter from './hooks';
+import { unwrap } from './utils/rxhelpers';
+import Button from '../components/common/Button';
 
-async function getShakeObservable(): Promise<Observable<number>> {
-  if (!window.DeviceMotionEvent) {
+enum MotionPermission {
+  NOT_SET = 0,
+  DENIED = 1,
+  GRANTED = 1
```

Comment:
Is there a reason the ints used to represent are the same? :O

Reply:
o shit
lol typo